api/app/tests/weather_models/test_env_canada_hrdps.py

```diff
@@ -99,13 +99,12 @@ def mock_requests_get_hrdps(*args, **kwargs):
 def test_get_hrdps_download_urls():
     """ test to see if get_download_urls methods gives the correct number of urls """
     # -1 because 000 hour has no APCP_SFC_0
-    total_num_of_urls = 49 * len(app.jobs.env_canada.GRIB_LAYERS) - 1
+    total_num_of_urls = 49 * len(app.jobs.env_canada.HRDPS_GRIB_LAYERS) - 1
     assert len(list(app.jobs.env_canada.get_high_res_model_run_download_urls(
         time_utils.get_utc_now(), 0))) == total_num_of_urls
 
 
-@pytest.mark.usefixtures('mock_get_processed_file_record')
-def test_process_hrdps(mock_download, mock_database):
+def test_process_hrdps(mock_download, mock_database, mock_get_processed_file_record):
```

Comment:
I have no idea. According to the docs, https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#use-fixtures-in-classes-and-modules-with-usefixtures `pytest.mark.usefixtures` gets used when declaring a test module using `class ...` so I don't understand why usefixtures() has been used this way in our other test files

Reply:
The annotation and the parameter should be the same but I'm not sure about the ordering. I think we should keep it as is until we figure that out and choose one way.

Reply:
We have a fixture ordering problem here. You can surface it by making all fixtures part of the annotation with:
`@pytest.mark.usefixtures('mock_get_processed_file_record', 'mock_download', 'mock_database')` and the function taking no parameters.

The mock used is the `mock_get_processed_file_record` within `mock_database`, not the one outside. 
The one we use expects that the `url` passed in matches `hrdps_url` to get the processing branch hit. The test fails because none of the urls pass in match it.